### DISCUSSION TITLE
Filter that ignores spaces

### DISCUSSION THREAD

**Discussion by EthanKC** [nextgenhealthcare/connect#5779] [General]
This is a dumb question, but I'm out of practice on my javascript, can't remember how to do this, and can't for the life of me find anything that jogs my memory. 
I'm trying to create a filter that only allows messages that have specific procedure codes through. The issue I ran into was that one of those codes has a space at the end. I can't use contains, and I want it to be more dynamic that just adding the value with a space as one of the variables. 
I'm trying to use trim, but can't get it to work. Here's what I have right now: 

```JS 
var string = msg['OBR']['OBR.4']['OBR.4.1'].toString()
var trimst = string.trim()

if (trimst == "G02043DB" || trimst == "G0204IMP3D" || trimst == "G02063DU" || trimst == "G0206IMP3D" || trimst == "G02034D" || trimst == "G02023D" || trimst == "G0202IM3D" || trimst == "G02023DUN" || trimst == "G02023DUIM" || trimst == "G0204I" || trimst == "G0204" || trimst == "G0206" || trimst == "G0206UNI" || trimst == "G0202I" || trimst == "G0202UNI" || trimst == "G0202UN" || trimst == "G020233D" || trimst == "G0202") {
    return true;
}
return false;
```

Obviously I'm currently trying with a javascript filter, but if there's a way I can add the trim function to a Rule Builder filter that would be even better.

**Comment by pacmano1** [comment]
Rule builders over time are worse in my opinion than just return true and false, like:

```JS
var my_codes = ['G02043DB','000296249']  // put your codes here
return my_codes.filter(code => code == msg['OBR']['OBR.4']['OBR.4.1'].toString().trim()).length > 0 // return true or false
```

**Reply by pacmano1** [reply]
Also, when you post code and want it to be snippet is three back ticks and "JS" and end it with three back ticks.

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/493e78ec-6121-4eda-b3ab-3bff837a8401)

**Reply by EthanKC** [reply]
Thanks, this helped! Also, after going to lunch and coming back, I realized the message wasn't being filtered by this rule I was working on, it was being filtered by the "accept if ORM" because it was an ORU...

**Reply by tonygermano** [reply]
No need for a javascript filter. Assuming you have a semi-recent version of mirth, you can just do
```js
var my_codes = ['G02043DB','000296249']  // put your codes here
return my_codes.includes(msg['OBR']['OBR.4']['OBR.4.1'].toString().trim())
```
